app/services/summary_service.rb

```diff
@@ -100,6 +120,21 @@ def self.add_arrival_dates(offenders)
     end
   end
 
+  def self.add_allocated_poms_and_coms(offenders, active_allocations_hash)
+    offenders.each do |offender|
+      next unless active_allocations_hash.key?(offender.offender_no)
+
+      alloc = active_allocations_hash[offender.offender_no]
+
+      # Add POM details
+      offender.allocated_pom_name = restructure_pom_name(alloc.primary_pom_name)
+      offender.allocation_date = (alloc.primary_pom_allocated_at || alloc.updated_at)&.to_date
+
+      # Add COM details
+      offender.allocated_com_name = alloc.com_name
```

Comment:
I'm not familiar with this part of the domain, please forgive my innocent question: Do COM names not need the same formatting as POM names?

Reply:
Good question. I'm not familiar with how these COM names are stored, so this is also a bit unclear to me. I haven't yet been able to get my hands on an offender object which has an COM assigned (I'm guessing this sort of data probably lives mostly on preprod or prod...?)

That said – our subject matter expert didn't flag any issues with how names are displaying when reviewing it. It seems like they might already have been 'restructured' elsewhere, or perhaps come in like that from the nDelius import.

Reply:
We'll soon notice this if it breaks, so np.